devolutions-gateway/src/jet_client.rs

```diff
@@ -73,6 +83,73 @@ impl JetClient {
     }
 }
 
+async fn handle_build_tls_proxy(
+    config: Arc<Config>,
+    response: HandleConnectJetMsgResponse,
+    interceptor: PcapRecordingInterceptor,
+    tls_acceptor: TlsAcceptor,
+) -> Result<(), io::Error> {
+    let client_stream = response.client_transport.get_tcp_stream();
+    let server_stream = response.server_transport.get_tcp_stream();
+
+    if client_stream.is_some() && server_stream.is_some() {
+        let tls_stream = tls_acceptor.accept(client_stream.unwrap()).await.map_err(|err| err)?;
+        let client_transport = TcpTransport::new_tls(TlsStream::Server(tls_stream));
+
+        let tls_handshake = create_tls_connector(server_stream.unwrap())
+            .await
+            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+        let server_transport = TcpTransport::new_tls(TlsStream::Client(tls_handshake));
+
+        Proxy::new(config)
+            .build_with_packet_interceptor(server_transport, client_transport, Some(Box::new(interceptor)))
+            .await
+    } else {
+        Err(io::Error::new(
+            io::ErrorKind::InvalidData,
+            "Failed to retrieve tcp stream to create tls connection!",
+        ))
+    }
+}
+
+async fn handle_build_proxy(
+    jet_associations: JetAssociationsMap,
+    config: Arc<Config>,
+    response: HandleConnectJetMsgResponse,
+    tls_acceptor: TlsAcceptor,
+) -> Result<(), io::Error> {
+    let mut recording_interceptor: Option<PcapRecordingInterceptor> = None;
+    let association_id = response.association_id;
+
+    let associations = jet_associations.lock().await;
+    if let Some(association) = associations.get(&association_id) {
+        if let Some(jet_tp_claim) = association.get_jet_tp_claim() {
+            if jet_tp_claim.eq(EXPECTED_JET_TP_VALUE) && config.plugins.is_some() {
```

Comment:
Similar to what we did in den-server, I think jet_tp_claim should be an enum with all possibles values. So instead of looking if we have a specific string, we could validate an enum

Reply:
done